File: web/vector.dart
/**
 * (c) 2016 Jörn Dinkla, www.dinkla.net
 * See the file LICENSE in the ROOT directory.
 *
 * Ein zweidimensionaler Vektor.
 */

import 'dart:math';

/**
 * A two dimensional vector.
 */
class Vector {

  final num x;
  final num y;

  const Vector(this.x, this.y);

  Vector operator +(Vector v) => new Vector(x + v.x, y + v.y);

  String toString() => "($x, $y)";

  @override
  bool operator ==(Vector v) => x == v.x && y == v.y;

  // if one overrides ==, you should also override hashCode
  @override
  int get hashCode => (x*41+y).hashCode;

  static const origin = const Vector(0, 0);

}

File: src/vector.js
// /**
//  * (c) 2016 Jörn Dinkla, www.dinkla.com
//  * See the file LICENSE in the ROOT directory.
//  *
//  * Ein zweidimensionaler Vektor.
//  */

// import 'dart:math';

// /**
//  * A two dimensional vector.
//  */
class Vector {

  constructor(x = 0.0, y = 0.0) {
    this.x = x;
    this.y = y;
  }

  plus(w) {
    return new Vector(this.x + w.x, this.y + w.y);
  }

  static origin = new Vector(0.0, 0.0);

}

export default Vector;
